controllers/swiftproxy_controller.go

```diff
@@ -286,6 +290,44 @@ func (r *SwiftProxyReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	password := string(sps.Data[instance.Spec.PasswordSelectors.Service])
 
+	//
+	// Check for required memcached used for caching
+	//
+	memcached, err := r.getSwiftMemcached(ctx, helper, instance)
```

Comment:
@cschwede I think we should move this block that requeue and trigger a reconciliation loop in the top-level controller.
The `memcached` instance is something shared between `proxy` and `storage` (they reference the same instance that is inherited from the top-level CR), so it makes sense wait or trigger a reconciliation loop once and at the [upper level](https://github.com/openstack-k8s-operators/swift-operator/blob/main/controllers/swift_controller.go#L293).
The result would be the same, but with a couple of advantages:
- code dedup, so we don't have to duplicate the very same logic
- in a greenfield you have the top-level CR waiting for the missing resource instead of creating the underlying resources

Reply:
Very good point! I just pushed the refactored PR.